### PR TITLE
Move mod/crepair to src/Module/Contact/Repair

### PR DESCRIPTION
**Pull request by nupplaphil**
Part of #4090

### CODE REVIEW

File: src/Module/Contact/Repair.php [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+		}
+
+		return;
+	}
+
+	public static function content(array $parameters = [])
+	{
+		$cid = $parameters['contact'];
+
+		$cid = $parameters['contact'];
```
Comment (MrPetovan) [inline src/Module/Contact/Repair.php]: Duplicated line.

File: src/Module/Contact/Repair.php [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+		$contact = Model\Contact::selectFirst([], ['id' => $cid, 'uid' => local_user()]);
+		if (empty($contact)) {
+			throw new BadRequestException(DI::l10n()->t('Contact not found.'));
+		}
+
+		if (empty(DI::page()['aside'])) {
```
Comment (MrPetovan) [inline src/Module/Contact/Repair.php]: It is initialized on line 42 of `src/App/Page.php` and therefore this initialization isn't necessary.

File: src/Module/Contact/Repair.php [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+			DI::page()['aside'] = '';
+		}
+
+		$a = DI::app();
+
+		$a->data['contact'] = $contact;
```
Comment (MrPetovan) [inline src/Module/Contact/Repair.php]: Do we really need this?

**Comment by annando** [comment]
I would like to call it differently. You also control the "remote-self" from there. So "advanced" is a better name.

**Comment by nupplaphil** [comment]
So `Module/Contact/Advanced` ?

File: src/Module/Contact/Repair.php [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+		$returnaddr = "contact/$cid";
+
+		$allow_remote_self = DI::config()->get('system', 'allow_users_remote_self');
+
+		// Disable remote self for everything except feeds.
+		// There is an issue when you repeat an item from maybe twitter and you got comments from friendica and twitter
+		// Problem is, you couldn't reply to both networks.
+		if (!in_array($contact['network'], [Protocol::FEED, Protocol::DFRN, Protocol::DIASPORA, Protocol::TWITTER])) {
+			$allow_remote_self = false;
```
Comment (MrPetovan) [inline src/Module/Contact/Repair.php]: This `if` is unnecessary and the whole thing can be just:
```php
$allow_remote_self = in_array($contact['network'], [Protocol::FEED, Protocol::DFRN, Protocol::DIASPORA, Protocol::TWITTER])
	&& DI::config()->get('system', 'allow_users_remote_self');
```

File: src/Module/Contact/Repair.php [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+		}
+
+		if ($contact['network'] == Protocol::FEED) {
+			$remote_self_options = ['0' => DI::l10n()->t('No mirroring'), '1' => DI::l10n()->t('Mirror as forwarded posting'), '2' => DI::l10n()->t('Mirror as my own posting')];
+		} else {
+			$remote_self_options = ['0' => DI::l10n()->t('No mirroring'), '2' => DI::l10n()->t('Mirror as my own posting')];
+		}
+
+		$update_profile = in_array($contact['network'], Protocol::FEDERATED);
```
Comment (MrPetovan) [inline src/Module/Contact/Repair.php]: Useless intermediate variable.

File: src/Module/Contact/Repair.php (hunk 3 above)
Comment (nupplaphil) [reply src/Module/Contact/Repair.php]: Doesn't seems like, I scrolled through `Profile::load()`

Comment (MrPetovan) [reply src/Module/Contact/Repair.php]: CHOP CHOP

**Pull request merged**